### PR TITLE
libstudxml: init at 1.1.0-b.10+2

### LINKED ISSUE #288575: libstudxml
libstudxml is a streaming XML pull parser and streaming XML serializer implementation for modern, standard C++. 
      It has an API that we believe should have already been in Boost or even in the C++ standard library.
      https://www.codesynthesis.com/projects/libstudxml/

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/288575

This PR adds 1 library: `libstudxml`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/li/libstudxml/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  outputs = [ "out" "dev" "doc" ];
+
+  nativeBuildInputs = [ build2 ];
+
+  meta = {
+    description = "A streaming XML pull parser and streaming XML serializer implementation for modern, standard C++";
+    homepage = "https://www.codesynthesis.com/projects/libstudxml/";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platforms = lib.platforms.all;
```
Comment (jtbx) [inline pkgs/by-name/li/libstudxml/package.nix:26]:
```suggestion
  meta = with lib; {
    description = "A streaming XML pull parser and streaming XML serializer implementation for modern, standard C++";
    homepage = "https://www.codesynthesis.com/projects/libstudxml/";
    license = licenses.mit;
    maintainers = with maintainers; [ tomasajt ];
    platforms = platforms.all;
```

Comment (TomaSajt) [reply pkgs/by-name/li/libstudxml/package.nix:26]: I'd rather not, sorry. I know it's widely used, but I like being more explicit. I usually only use `with` right before lists.

Comment (jtbx) [reply pkgs/by-name/li/libstudxml/package.nix:26]: Fair enough, I'm actually seeing more of this type of thing these days :)

**Comment by jtbx** [comment]
Result of `nixpkgs-review pr 288616` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>libstudxml</li>
    <li>libstudxml.dev</li>
    <li>libstudxml.doc</li>
  </ul>
</details>

**Review by jtbx — APPROVED** [review]

**Comment by pbsds** [comment]
The files in /lib/ are not marked executable

**Comment by TomaSajt** [comment]
> The files in /lib/ are not marked executable
Fixed.

**Pull request merged**